src/Route/RouteCollection.php

```diff
@@ -84,6 +84,10 @@ public function add(
 
         if (!isset($defaults['_controller'])) {
             $actionJoiner = false === \strpos($this->baseControllerName, '\\') ? ':' : '::';
+            if (false !== \strpos($this->baseControllerName, '\\') && false !== \strpos($this->baseControllerName, ':')) {
```

Comment:
I don't like having 2 same checks `false !== \strpos($this->baseControllerName, '\\')`

Reply:
What is your suggestion?

Reply:
Maybe do:

```
if ($actionJoiner !== ':' && ...) {
```